src/libraries/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs

```diff
@@ -199,7 +199,31 @@ public string Category
         /// <summary>
         /// Looks up the localized name of a given category.
         /// </summary>
-        protected virtual string? GetLocalizedString(string value) => SR.GetResourceString("PropertyCategory" + value);
+        protected virtual string? GetLocalizedString(string value) => value switch
+        {
+            "Action" => SR.PropertyCategoryAction,
+            "Appearance" => SR.PropertyCategoryAppearance,
+            "Asynchronous" => SR.PropertyCategoryAsynchronous,
+            "Behavior" => SR.PropertyCategoryBehavior,
+            "Config" => SR.PropertyCategoryConfig,
+            "Data" => SR.PropertyCategoryData,
+            "DDE" => SR.PropertyCategoryDDE,
+            "Default" => SR.PropertyCategoryDefault,
+            "Design" => SR.PropertyCategoryDesign,
+            "DragDrop" => SR.PropertyCategoryDragDrop,
+            "Focus" => SR.PropertyCategoryFocus,
+            "Font" => SR.PropertyCategoryFont,
+            "Format" => SR.PropertyCategoryFormat,
+            "Key" => SR.PropertyCategoryKey,
+            "Layout" => SR.PropertyCategoryLayout,
+            "List" => SR.PropertyCategoryList,
+            "Mouse" => SR.PropertyCategoryMouse,
+            "Position" => SR.PropertyCategoryPosition,
+            "Scale" => SR.PropertyCategoryScale,
+            "Text" => SR.PropertyCategoryText,
+            "WindowStyle" => SR.PropertyCategoryWindowStyle,
+            _ => SR.GetResourceString("PropertyCategory" + value, value)
```

Comment:
Seems like adding a default value on the default case is breaking some unit tests as this is actually changing the behavior for custom categories. I will instead change the default case to not have a default value